FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt

```diff
@@ -184,9 +184,25 @@ data class OrgRules constructor(
    * public keys for these domains, such as from their own internal keyserver
    */
   fun canLookupThisRecipientOnAttester(emailAddr: String): Boolean {
-    val domain = EmailUtil.getDomain(emailAddr)
-    val domains = disallowAttesterSearchForDomains ?: emptyList()
-    return !domains.contains(if (domain.isEmpty()) "NONE" else domain)
+    if (disallowLookupOnAttester()) {
+      return false
+    }
+
+    val disallowedDomains = disallowAttesterSearchForDomains ?: emptyList()
+    val userDomain = EmailUtil.getDomain(emailAddr)
+    if (userDomain.isEmpty()) {
+      throw IllegalStateException("Not a valid email $emailAddr")
+    }
+
+    return !disallowedDomains.contains(userDomain)
```

Comment:
doesn't this require sort of case-insensitive comparison?

Reply:
This realization is based on https://github.com/FlowCrypt/flowcrypt-browser/blob/48cb02a244b71a2acffbbc0fdb6ed4254baa0da3/extension/js/common/org-rules.ts#L170-L192.

Does `include` mean case-insensitive comparison?

Reply:
In typescript we have:
```typescript
  public canLookupThisRecipientOnAttester = (emailAddr: string): boolean => {
    ....
    const userDomain = Str.getDomainFromEmailAddress(emailAddr);
    if (!userDomain) {
      throw new Error(`Not a valid email ${emailAddr}`);
    }
    return !disallowedDomains.includes(userDomain);
  }
```
and then:
```typescript
  public static getDomainFromEmailAddress = (emailAddr: string) => {
    // todo: parseEmail()?
    return emailAddr.toLowerCase().split('@')[1];
  }
```
So function for getting domain in the typescript casts it to lowercase and then if you configure everything in lowercase, all works.
In Kotlin that doesn't happen:
```kotlin
    // from EmailUtil.kt
    fun getDomain(email: String): String {
      return when {
        TextUtils.isEmpty(email) -> ""
        email.contains("@") -> email.substring(email.indexOf('@') + 1)
        else -> ""
      }
    }
```

Reply:
I see. Thank you. I will fix it.